openedx/features/course_experience/views/course_home.py

```diff
@@ -113,9 +114,12 @@ def render_to_fragment(self, request, course_id=None, **kwargs):
 
         # Render the full content to enrolled users, as well as to course and global staff.
         # Unenrolled users who are not course or global staff are given only a subset.
-        is_enrolled = CourseEnrollment.is_enrolled(request.user, course_key)
-        is_staff = has_access(request.user, 'staff', course_key)
-        if is_enrolled or is_staff:
+        user_access = {
```

Comment:
I like this concept and it seems that it could be useful to other views. I wonder if it would be worth having a helper method. You don't have to do that now, but we could consider it if a similar pattern is needed in the future.